### DISCUSSION TITLE
SSL Error

### DISCUSSION THREAD

**Discussion by Charlydt** [rspamd/rspamd#3942] [Q&A]
Hi team,

I have rspamd version 1.6.5 installed on a production server and I get below errors on the web interface:

error reading https://rspamd.com/freemail/free.txt.zst(135.181.136.158:443): connection with http server terminated incorrectly: ssl connect error: ssl error: error:14090086:SSL routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed
-
error reading https://rspamd.com/freemail/disposable.txt.zst(135.181.136.158:443): connection with http server terminated incorrectly: ssl connect error: ssl error: error:14090086:SSL routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed
-
error reading https://rspamd.com/phishtank/online-valid.json.zst(135.181.136.158:443): connection with http server terminated incorrectly: ssl connect error: ssl error: error:14090086:SSL routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed
-

I have noticed that the repository for the newer versions is maps.rspamd.com instead of rspamd.com. Will updating the maps url manually fix these errors? We are not planning to upgrade Rspamd at the moment and looking for a possible solution to this.

Any advice would be highly appreciated.

**Comment by rifnet-win19-65** [comment]
Hi, 

I am having the same issue on maps.rspamd.com. Seems like a certificate is expired. 

openssl s_client -connect maps.rspamd.com:443 
CONNECTED(00000003)
depth=3 O = Digital Signature Trust Co., CN = DST Root CA X3
verify error:num=10:certificate has expired
notAfter=Sep 30 14:01:15 2021 GMT
---
Certificate chain
 0 s:/CN=maps.rspamd.com
   i:/C=US/O=Let's Encrypt/CN=R3
 1 s:/C=US/O=Let's Encrypt/CN=R3
   i:/C=US/O=Internet Security Research Group/CN=ISRG Root X1
 2 s:/C=US/O=Internet Security Research Group/CN=ISRG Root X1
   i:/O=Digital Signature Trust Co./CN=DST Root CA X3
---
Server certificate

**Reply by m-i-k-e-y** [reply]
@rifnet-win19-65 
You're having a different issue. One of the root certificates that are used by Let's Encrypt to sign the certificates issued by Let's encrypt has expired on September 30th. See here for details: [https://letsencrypt.org/docs/dst-root-ca-x3-expiration-september-2021/](https://letsencrypt.org/docs/dst-root-ca-x3-expiration-september-2021/). Seems like you're OS is not up to date to handle this situation.

Connecting to maps.rspamd.com is working fine from an up to date Debian 11:

> $ uname -a && openssl s_client -connect maps.rspamd.com:443
Linux [redacted] 5.10.0-9-amd64 #1 SMP Debian 5.10.70-1 (2021-09-30) x86_64 GNU/Linux
CONNECTED(00000003)
depth=2 C = US, O = Internet Security Research Group, CN = ISRG Root X1
verify return:1
depth=1 C = US, O = Let's Encrypt, CN = R3
verify return:1
depth=0 CN = maps.rspamd.com
verify return:1

and an up to date Ubuntu 20.04:
> $ uname -a && openssl s_client -connect maps.rspamd.com:443
Linux [redacted] 5.4.0-88-generic #99-Ubuntu SMP Thu Sep 23 17:29:00 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
CONNECTED(00000003)
depth=2 C = US, O = Internet Security Research Group, CN = ISRG Root X1
verify return:1
depth=1 C = US, O = Let's Encrypt, CN = R3
verify return:1
depth=0 CN = maps.rspamd.com
verify return:1

**Reply by rifnet-win19-65** [reply]
Hi m-i-k-e-y,

Thanks! Added new CA root. 
It is an old centos 7 system, here's how to do it (wo updating openssl), maybe someone could need this:

```
Install the ca-certificates package: yum install ca-certificates
Enable the dynamic CA configuration feature: update-ca-trust force-enable
Cp new ca crt file (from [https://letsencrypt.org/certificates/]) to /etc/pki/ca-trust/source/anchors/
update-ca-trust extract

```
